File: app/controllers/api/v2/environments_controller.rb
#
# Copyright 2013 Red Hat, Inc.
#
# This software is licensed to you under the GNU General Public
# License as published by the Free Software Foundation; either version
# 2 of the License (GPLv2) or (at your option) any later version.
# There is NO WARRANTY for this software, express or implied,
# including the implied warranties of MERCHANTABILITY,
# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
# have received a copy of GPLv2 along with this software; if not, see
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.


class Api::V2::EnvironmentsController < Api::V1::EnvironmentsController

  include Api::V2::Rendering

  api :GET, "/organizations/:organization_id/environments/systems_registerable", "List environments that systems can be registered to"
  param :organization_id, :identifier, :desc => "organization identifier"
  def systems_registerable
    @environments = KTEnvironment.systems_registerable(@organization)
    respond_for_index :collection => @environments
  end

end

File: engines/bastion/app/assets/bastion/environments/environments.controller.js
/**
 * Copyright 2013 Red Hat, Inc.
 *
 * This software is licensed to you under the GNU General Public
 * License as published by the Free Software Foundation; either version
 * 2 of the License (GPLv2) or (at your option) any later version.
 * There is NO WARRANTY for this software, express or implied,
 * including the implied warranties of MERCHANTABILITY,
 * NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
 * have received a copy of GPLv2 along with this software; if not, see
 * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
 */

/**
 * @ngdoc object
 * @name  Bastion.environments.controller:EnvironmentsController
 *
 * @requires $scope
 * @requires $timeout
 * @requires $http
 * @requires Routes
 * @requires CurrentOrganization
 *
 * @description
 *   Provides the functionality for the environments path page.
 */
angular.module('Bastion.environments').controller('EnvironmentsController',
    ['$scope', '$timeout', '$http', 'Routes', 'CurrentOrganization',
        function($scope, $timeout, $http, Routes, CurrentOrganization) {

            $scope.successMessages = [];
            $scope.errorMessages = [];

            if (!$scope.environmentsTable) {
                $scope.environmentsTable = {rows: []};
            }

            $http.get(Routes.apiOrganizationEnvironmentsPath(CurrentOrganization) + '/paths')
            .success(function(paths) {
                $scope.environmentsTable.rows = paths;
            });

            $scope.initiateCreatePath = function () {
                if ($scope.environmentsTable.rows[0].path.length > 1) {
                    $scope.environmentsTable.rows.unshift([]);
                    $scope.environmentsTable.rows[0].path = [$scope.environmentsTable.rows[1].path[0]];
                    $scope.environmentsTable.rows[0].permissions = $scope.environmentsTable.rows[1].permissions;
                }
                $scope.environmentsTable.rows[0].pathId = 0;
                $scope.environmentsTable.rows[0].showCreate = true;
            };

            $scope.readonly = function() {
                if ($scope.environmentsTable.rows.length > 0) {
                    return $scope.environmentsTable.rows[0].permissions.readonly;
                } else {
                    return false;
                }
            };
        }]
);
